src/order/conditionally_complete_lattice.lean

```diff
@@ -133,9 +133,8 @@ section
 open_locale classical
 
 /-- A well founded linear order is conditionally complete, with a bottom element. -/
-@[reducible] noncomputable def well_founded.conditionally_complete_linear_order_with_bot
-  {α : Type*} [i : linear_order α] (h : well_founded ((<) : α → α → Prop))
-  (c : α) (hc : c = h.min set.univ ⟨c, mem_univ c⟩) :
+@[reducible] noncomputable def well_founded.conditionally_complete_linear_order_bot
+  (α : Type*) [i : linear_order α] [h : is_well_order α (<)] [b : order_bot α] :
```

Comment:
Note this PR goes one step further than #14568 by switching to `is_well_order`; so is possibly still worth having. Let's wait for the other PR to merge, then get @urkud's opinion.

Reply:
I'd argue that `linear_order α` + `is_well_order α (<)` is the current `mathlib` spelling for "α is a well-order", and should be universally preferred over `linear_order α` + `well_founded (<)`, which is equivalent but doesn't work with typeclass inference.